include/arch/arm/arch/64/mode/machine/registerset.h

```diff
@@ -291,5 +314,25 @@ static inline void Arch_initContext(user_context_t *context)
 #endif
 }
 
+/* Definitions for inlinle assembly */
+#if __has_feature(capabilities)
```

Comment:
General remark:

Sometimes you use __has_features, sometimes you use defines. I suggest only using the defines, and perhaps defining one of them to `__has_feature(capabilities)`. But avoid using that outside of that define, as in seL4's context which has its own different kind of capabilities, it's confusing.

Reply:
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005ffeature.html

Reply:
I think what Indan means is just a macro on top of `__has_feature`, e.g.

```c
#if __has_feature(capabilities)
#define SEL4_CHERI 1
#endif
```

or something like that (maybe a better name than that). This brings it in line with the rest of the `#ifdef` statements and leave some flexibility for changing things based on compiler, config, etc.

Reply:
Yeah that makes sense. I'll work on it